src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java

```diff
@@ -271,4 +294,26 @@ private void loadMetricsDBFilesCleanupEnabled() {
       shouldCleanupMetricsDBFiles = true;
     }
   }
+
+  private void loadBatchMetricsRetentionPeriodFromConfig() {
+    if (!settings.containsKey(BATCH_METRICS_RETENTION_PERIOD)) {
+      return;
+    }
+
+    long parsedRetentionPeriod;
+    try {
+      parsedRetentionPeriod = Long.parseLong(settings.getProperty(BATCH_METRICS_RETENTION_PERIOD));
+      if (parsedRetentionPeriod <= 0 || parsedRetentionPeriod > BATCH_METRICS_RETENTION_PERIOD_LIMIT) {
+        throw new InvalidParameterException();
+      }
+      batchMetricsRetentionPeriod = parsedRetentionPeriod;
+    } catch (NumberFormatException | InvalidParameterException e) {
+      LOG.error(
+              (Supplier<?>)
+                      () ->
+                              new ParameterizedMessage(
+                                      "Invalid batch-metrics-retention-period. Using default value {}.", batchMetricsRetentionPeriod),
+              e);
```

Comment:
same here. Log error and then we should just use the defaults.

Also you can log error simply as:

`LOG.error("my error message: {}", myParam)`

Reply:
Agreed, updated the error logging call.